### ISSUE TITLE
[FEATURE] Support for MSVC

### ISSUE THREAD

**Issue by jfalcou** [jfalcou/eve#1236]
This is a mutable, ongoing issue to discuss/list steps for proper MSVC support.

# Current Status
Since MSVC 2022 19.30, C++20 support has drastically improved. Building blocks libraries like kumi and raberu now natively compile on MSVC without any hassle. This has triggered the investigation of how much EVE needs to be fixed to make MSVC EVE a thing.

Note that we currently don't care about code generation quality as our first goal here is to get the library to compile.

# Plan of Actions
The first batch of trivial fixes has been PR'ed in #1233 and fixed some glaring issues with fundamental code.
We'll list there next steps to be taken.

# Current Status
- `unit.arch.exe` compiles and pass tests

**Comment by DenisYaroshevskiy** [comment]
CI?

I'll have to install windows somewhere, won't I?

**Comment by jfalcou** [comment]
Was thinking of using appveyor

**Comment by Tradias** [comment]
Is there anything you need help with here?

Apart from appveyor, I have only ever used Github Actions and they were easy to set up.

**Comment by jfalcou** [comment]
I just need to find time to tackle the stupid errors that prevent other test to compile.

 Appveyor is setup already.

**Comment by jfalcou** [comment]
There looks like some compile time trials we have don't fly properly but the error messages are outrageous.

I  an push a more up-to-date list of breaking thing later today.

**Comment by jfalcou** [comment]
So mostly we need help understanding why msvc don't understand our properly standard templatr monstrosity

**Comment by Tradias** [comment]
Sure, sounds like something I can investigate.

**Comment by jfalcou** [comment]
Good. Let me update this issue later withore info

**Comment by jfalcou** [comment]
@Tradias added a small roadmap on how to tackle the thing. Tell me if you need help.

**Comment by jfalcou** [comment]
# unit.api targets
 - Looks like we try to use constepxr value as lambda capture in some of the shuffle tests and MSVC doesn't appreciate that

**Comment by Tradias** [comment]
Looks like, I have managed to get `unit.core.exe` to compile without significant code changes. Creating a pr now.

**Comment by jfalcou** [comment]
Good job wow :DDid you used the in flight PR that fixes API ? or were the issue unrelated ?

**Comment by Tradias** [comment]
I didn't know about https://github.com/jfalcou/eve/pull/1458, one of the issue was in fact related. I think my solution might be faster to compile but potentially incorrect.

**Comment by jfalcou** [comment]
I had the same code at one point but it failed on other compilers. Maybe we can borrow your cocnept based strategy but lay  it out like ours ?

**Comment by Tradias** [comment]
Yes that is possible. I have reverted it to your version for now.

**Comment by jfalcou** [comment]
Merged the CI adaptation to see if it is faster to compiles
